### DISCUSSION TITLE
building the installer

### DISCUSSION THREAD

**Discussion by ianamk** [sumatrapdfreader/sumatrapdf#2753] [General]
Hello 
I see the current sumatrapdf comes in a separate installer like it did in the past instead of carrying the dlls inside it self, my question is how to build the installer in the past there was an installer project file now there isn't and one cant be generated with premake5 how do you compile the exe from visual studio.

**Comment by kjk** [comment]
The installer is the same executable as `SumatraPDF-dll` project.

It shows the installation / uninstallation UI when it auto-detects that it's running as an installer (an exe name contains `install` or given explicit flags `-install`, `-uninstall`).

**Reply by ianamk** [reply]
wow didn't notice that thank you I thought it was a separate project
